source/common/event/file_event_impl.cc

```diff
@@ -24,6 +24,7 @@ FileEventImpl::FileEventImpl(DispatcherImpl& dispatcher, os_fd_t fd, FileReadyCb
               ? Runtime::runtimeFeatureEnabled(
                     "envoy.reloadable_features.activate_fds_next_event_loop")
               : true) {
+  RELEASE_ASSERT(SOCKET_VALID(fd), "");
```

Comment:
I still think the comment is valuable, in some form, but will defer to @antoniovicente on that.

Reply:
Keeping the comment SGTM.

Reply:
Done!